xmldiff/diff.py

```diff
@@ -162,23 +162,31 @@ def node_ratio(self, left, right):
             # One is a comment the other is not:
             return 0
 
+        match = None
         for attr in self.uniqueattrs:
-            if not isinstance(attr, str):
-                # If it's actually a sequence of (tag, attr), the tags must
-                # match first.
-                tag, attr = attr
-                if tag != left.tag or tag != right.tag:
-                    continue
-            if attr in left.attrib or attr in right.attrib:
-                # One of the nodes have a unique attribute, we check only that.
+            if isinstance(attr, dict):
+                for tag, attr_list in attr.items():
+                    if tag != left.tag or tag != right.tag:
+                        continue
+                    matches = 0
+                    attr_list_consolidated = \
+                        [attr for attr in attr_list
+                         if attr in left.attrib or attr in right.attrib]
+                    for inner_attr in attr_list_consolidated:
+                        matches += int(left.attrib.get(inner_attr)
+                                       == right.attrib.get(inner_attr))
```

Comment:
You could also shortcut this here. You don't need to check all attributes unless the match. Find one that doesn't match, and it's not a match. So no need to count matches.